s2i-liberty/readme.md

```diff
@@ -0,0 +1,197 @@
+```
+Second modify the newly created build config to look as follows:
+```
+    sourceStrategy:
+      from:
+        kind: ImageStreamTag
+        name: 'eap70-openshift:latest'
+      runtimeImage:
+        kind: ImageStreamTag
+        name: 'liberty-runtime-s2i:latest'
+      runtimeArtifacts:
+        - sourcePath: /opt/eap/standalone/deployments/hello-world-war-1.0.0.war
+          destinationDir: artifacts
+```
+notice that the source strategy has two additional sections:
+* `runtimeImage`: which defines the runtime image to be used during the build
+* `runtimeArtifacts:`: which is an array and specifies a list of artifacts that will be copied from the builder image to the runtime image.
+
+For mode details on how this copy process works, see the [runtime image documentation](https://github.com/openshift/source-to-image/blob/master/docs/runtime_image.md).
+
+The liberty runtime image expects:
+* any deployable artifacts in the $WORKDIR/artifacts directory
+* any [Liberty configuration files] (http://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.doc/ae/cwlp_config.html) in the $WORKDIR/config directory 
+
+As you can notice the runtimeArtifacts section mentions files by name. This implies that for this process to work the artifact created by the builder image have to be always the same. I expect improvements on this front in future releases of OpenShigt but for now this is a constraint that must be kept in mind.
+
+### Create a new Application
+
+To demonstrate the usage of the newly created builder and runtime images, a Jee example application will be built and deployed to Liberty using the Source to Image process. 
+
+Create the new application by passing in the name of the newly created and configured extended builder image :
+
+```
+oc new-app -i hello-world --name=hello-world
+```
+
+Let's break down the command in further detail
+
+* `oc new-app` - OpenShift command to create a a new application
+* `-i=hello-world` - Name of the ImageStream that contains the result of the build config that uses the extended s2i process
+* `--name=play-app` - Name to be applied to the newly created resources
+
+The build that was triggered by the `new-app` command can be found by executing the following command:
+
+```
+oc get builds -l=app=play-app
+```
+
+View the build logs by executing the following command:
+
+```
+oc logs builds/<build_name>
+```
+
+*Note: Replace `<build_name>` with the name of the build found in the previous command.*
+
+Once the build completes, the application will be deployed.
+
+## Considerations on http session failover
+
+When liberty is deployed in a cloud environment certain limitation applies as explained in this [document](http://www.ibm.com/support/knowledgecenter/en/SSD28V_8.5.5/com.ibm.websphere.wlp.core.doc/ae/cwlp_paas_restrict.html).
```

Comment:
Sentence needs to be cleaned up. "When liberty is deployed in a cloud environment, there are certain limitations that apply as explained in this document"